### PR TITLE
Sowmya | BAH-3313 | To add Nursing Tasks display control in new IPD dashboard with the ability to display pending and late (and not administered yet) medication tasks

### PR DESCRIPTION
**Pull request by sowmya-AS**
https://bahmni.atlassian.net/browse/BAH-3313

### CODE REVIEW

File: src/features/DisplayControls/NursingTasks/components/NursingTasks.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        <div key={index}>
+          <TaskTile medicationNursingTask={medicationNursingTask} />
+        </div>
+      );
+    });
+  };
+
+  const showMedicationNursingTasks = () => {
+    if (isLoading) {
+      return <div>Loading...</div>;
```
Comment (Arjun-Go) [inline src/features/DisplayControls/NursingTasks/components/NursingTasks.jsx:62]: Can we use a Skeleton over here - https://react.carbondesignsystem.com/?path=/docs/components-datatable-skeleton--overview

File: src/utils/config.json [hunk 2]
```diff
@@ -25,7 +25,16 @@
...
+      },
+      {
+        "name": "Nursing Tasks",
+        "component": "NT",
+        "displayOrder": 6
       }
-    ]
+    ],
+    "nursingTasks": {
+      "timeInMinutesFromNowToShowTaskAsRelevant": 60,
```
Comment (Arjun-Go) [inline src/utils/config.json:36]: Do we keep it here or keep it as part of a config. where is this getting used?

Comment (sowmya-AS) [reply src/utils/config.json:36]: all the configs for this repo as per previous discussion , for now would reside within config.json

File: src/entries/Dashboard/componentMapping.js [hunk 3]
```diff
@@ -10,4 +10,9 @@ export const componentMapping = {
   AL: lazy(() =>
     import("../../features/DisplayControls/Allergies/components/Allergies")
   ),
+  NT: lazy(() =>
+    import(
+      "../../features/DisplayControls/NursingTasks/components/NursingTasks"
```
Comment (ayush34957) [inline src/entries/Dashboard/componentMapping.js:15]: format the document with prettier, please check for other files too

File: src/features/DisplayControls/NursingTasks/components/NursingTasks.jsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+  const NoNursingTasksMessage = (
+    <FormattedMessage
+      id={"NO_NURSING_TASKS_MESSAGE"}
+      defaultMessage={"No nursing task is scheduled for the patient"}
+    />
+  );
+
+  useEffect(() => {
+    const fetchNursingTasks = async () => {
```
Comment (ayush34957) [inline src/features/DisplayControls/NursingTasks/components/NursingTasks.jsx:26]: can you please move this method out of useEffect, just have the invoke inside useEffect

File: src/features/DisplayControls/NursingTasks/components/NursingTasks.jsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    };
+
+    fetchNursingTasks();
+  }, []);
+
+  const showCurrentDate = () => {
+    var currentDate = new Date();
+
+    var formattedDate = currentDate.toLocaleDateString();
+    return <div className="date-text">{formattedDate}</div>;
```
Comment (ayush34957) [inline src/features/DisplayControls/NursingTasks/components/NursingTasks.jsx:47]: we have the date utility methods in src/utils... check if we can reuse it

File: src/features/DisplayControls/NursingTasks/styles/TaskTile.scss [hunk 6]
```diff
@@ -0,0 +1,65 @@
+.tile {
```
Comment (ayush34957) [inline src/features/DisplayControls/NursingTasks/styles/TaskTile.scss:1]: can we change the class name? it is too generic and might have the same class name in Bahmni app

**Review by ayush34957 — APPROVED** [review]

**Pull request merged**